File: src/test/java/org/superbiz/moviefun/albums/AlbumTest.java
package org.superbiz.moviefun.albums;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

public class AlbumTest {

    @Test
    public void testIsEquivalent() {
        Album persisted = new Album("Radiohead", "OK Computer", 1997, 8);
        persisted.setId(10);

        Album sameFromCsv = new Album("Radiohead", "OK Computer", 1997, 9);
        assertThat(persisted.isEquivalent(sameFromCsv), is(true));

        Album otherFromCsv = new Album("Radiohead", "Kid A", 2000, 9);
        assertThat(persisted.isEquivalent(otherFromCsv), is(false));
    }
}

File: components/albums/src/test/java/org/superbiz/moviefun/albums/AlbumTest.kt
package org.superbiz.moviefun.albums

import org.junit.Test

import org.hamcrest.core.Is.`is`
import org.junit.Assert.assertThat

class AlbumTest {

    @Test
    fun testIsEquivalent() {
        val persisted = Album(id = 10, artist = "Radiohead", title = "OK Computer", year = 1997, rating = 8)

        val sameFromCsv = Album(artist = "Radiohead", title = "OK Computer", year = 1997, rating = 9)
        assertThat(persisted.isEquivalent(sameFromCsv), `is`(true))

        val otherFromCsv = Album(artist = "Radiohead", title = "Kid A", year = 2000, rating = 9)
        assertThat(persisted.isEquivalent(otherFromCsv), `is`(false))
    }
}
